Extend documentation with RegisterOrReplace api for behaviors and connectors (#5039)
Particular/docs.particular.net: nservicebus/pipeline/manipulate-with-behaviors_registerorreplace_core_[8.0,).partial.md

﻿## Add or replace a step

> The `Register` API will fail when a behavior is registered that is already present in the pipeline.
> On the counter side, the `Replace` API will fail when a behavior is registered that is not found in the pipeline.

Reviewer: ```suggestion
The `Register` API will throw when a behavior with the same ID that is already present in the pipeline while the `Replace` API will throw when a behavior with the specified ID is **not** found in the pipeline.
```

> The `Register` API will fail when a behavior is registered that is already present in the pipeline.
> On the counter side, the `Replace` API will fail when a behavior is registered that is not found in the pipeline.

Author: ```suggestion
The `Register` API will throw when a behavior is registered with an ID that that is already present in the pipeline.
On the counter side, the `Replace` API will throw when a behavior is registered with an ID that is not found in the pipeline.
```
